src/coreclr/jit/valuenum.cpp

```diff
@@ -7954,20 +7946,56 @@ void Compiler::fgValueNumberTreeConst(GenTree* tree)
 
 #ifdef FEATURE_SIMD
         case TYP_SIMD8:
+        {
+            simd8_t simd8Val;
+
+            if (tree->IsIntegralConst(0))
+            {
+                simd8Val = {};
+            }
+            else
+            {
+                simd8Val = tree->AsVecCon()->gtSimd8Val;
+            }
+
+            tree->gtVNPair.SetBoth(vnStore->VNForSimd8Con(simd8Val));
+            break;
+        }
+
         case TYP_SIMD12:
         case TYP_SIMD16:
+        {
+            simd16_t simd16Val;
+
+            if (tree->IsIntegralConst(0))
```

Comment:
It would be very nice to get rid of these `CNS_INT SIMD` nodes. They're only produced in `fgMorphPromoteLocalInitBlock`, it should be simple to switch that to using proper `CNS_VEC` constants.

Reply:
IIRC, I had tried that initially and believe I ran into issues so I was going to leave it towards a "future cleanup".

Mostly just trying to ensure the "core" work (the xplat hwintrinsic APIs) that is part of the planned feature set for  .NET 7  is unblocked immediately speaking.

Reply:
> IIRC, I had tried that initially and believe I ran into issues

That's interesting; I wonder what sorts of issues.

I am fine with leaving this as-is, but there should be a `TODO-Cleanup` about deleting this redundant IR shape.